client/reader/utils.js

```diff
@@ -16,40 +15,42 @@ export function isPostNotFound( post ) {
 }
 
 export function showSelectedPost( { replaceHistory, postKey, comments } ) {
-	if ( ! postKey ) {
-		return;
-	}
+	return ( dispatch, getState ) => {
+		if ( ! postKey ) {
+			return;
+		}
 
-	// rec block
-	if ( postKey.isRecommendationBlock ) {
-		return;
-	}
+		// rec block
+		if ( postKey.isRecommendationBlock ) {
+			return;
+		}
 
-	const post = getPostByKey( reduxGetState(), postKey );
+		const post = getPostByKey( getState(), postKey );
 
-	if ( isXPost( post ) && ! replaceHistory ) {
-		return showFullXPost( XPostHelper.getXPostMetadata( post ) );
-	}
+		if ( isXPost( post ) && ! replaceHistory ) {
+			return showFullXPost( XPostHelper.getXPostMetadata( post ) );
+		}
 
-	// normal
-	let mappedPost;
-	if ( postKey.feedId ) {
-		mappedPost = {
-			feed_ID: postKey.feedId,
-			feed_item_ID: postKey.postId,
-		};
-	} else {
-		mappedPost = {
-			site_ID: postKey.blogId,
-			ID: postKey.postId,
-		};
-	}
+		// normal
+		let mappedPost;
+		if ( postKey.feedId ) {
+			mappedPost = {
+				feed_ID: postKey.feedId,
+				feed_item_ID: postKey.postId,
+			};
+		} else {
+			mappedPost = {
+				site_ID: postKey.blogId,
+				ID: postKey.postId,
+			};
+		}
 
-	showFullPost( {
-		post: mappedPost,
-		replaceHistory,
-		comments,
-	} );
+		showFullPost( {
+			post: mappedPost,
+			replaceHistory,
```

Comment:
I tried to lookup when and how is the `replaceHistory` option used, and didn't find any usage. It's always `undefined`. The last usage was removed when Reduxifying Reader in #22814. Would be nice to do an extra commit that removes it.

Reply:
Good find 👍  Cleaned it up in https://github.com/Automattic/wp-calypso/pull/63954/commits/d54b7018f4d355603f90587359b4c312213aac71